debezium-ai/debezium-ai-docling/src/main/java/io/debezium/ai/docling/FieldToDocling.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.ai.docling;
+
+import static java.lang.String.format;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.components.Versioned;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.Module;
+import io.debezium.config.Configuration;
+import io.debezium.config.EnumeratedValue;
+import io.debezium.config.Field;
+import io.debezium.transforms.ConnectRecordUtil;
+import io.debezium.transforms.SmtManager;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+import ai.docling.api.serve.DoclingServeApi;
+import ai.docling.api.serve.convert.request.ConvertDocumentRequest;
+import ai.docling.api.serve.convert.request.options.ConvertDocumentOptions;
+import ai.docling.api.serve.convert.request.options.InputFormat;
+import ai.docling.api.serve.convert.request.options.OutputFormat;
+import ai.docling.api.serve.convert.request.source.FileSource;
+import ai.docling.api.serve.convert.request.source.HttpSource;
+import ai.docling.api.serve.convert.request.source.Source;
+import ai.docling.api.serve.convert.request.target.InBodyTarget;
+import ai.docling.api.serve.convert.response.ConvertDocumentResponse;
+import ai.docling.client.serve.DoclingServeClientBuilderFactory;
+
+/**
+ * Single message transform which appends to the record <a href="https://github.com/docling-project/">Docling</a> transformation of selected {@link String} field
+ * or replace this field with it.
+ * Docling is able to parse various kinds of inpout formats and convert them to selected output format, making it more easy and efficient to consume these documents
+ * by LLMs and AI in general.
+ *
+ * @author vjuranek
+ */
+public class FieldToDocling<R extends ConnectRecord<R>> implements Transformation<R>, Versioned {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FieldToDocling.class);
+
+    private static final Schema DOCLING_SCHEMA = Schema.STRING_SCHEMA;
```

Comment:
I wonder if this should be just plain String or a logical type with either custom name or a parameter reflecting this is docling and its format is